### PR TITLE
[Core] Simplify multimodal masking

### PR DESCRIPTION
**Pull request by lgeiger**
## Purpose

Since PyTorch 2.9.0 (https://github.com/pytorch/pytorch/pull/156384) `target[mask] = src` doesn't cause `cudaStreamSynchronize` anymore in cases where `mask` is a CPU tensor.

This PR simplifies `_merge_multimodal_embeddings` by removing the need for `masked_scatter_` without re-introducing a CPU/GPU sync. This also simplifies the model runner since the mask doesn't need to be transferred to the GPU anymore.

## Test Plan

I verified that for Qwen3VL no `cudaStreamSynchronize` ops are visible in the torch profile.

I'm keeping this as a draft for now to see what @DarkLight1337 or @ywang96 think of this approach. I'm happy to then also double check some other multi modal models.

```bash
vllm serve Qwen/Qwen3-VL-2B-Instruct-FP8 --limit-mm-per-prompt.video 0 --gpu-memory-utilization 0.96

vllm bench serve --backend openai-chat --model Qwen/Qwen3-VL-2B-Instruct-FP8 --endpoint /v1/chat/completions --dataset-name hf --dataset-path lmarena-ai/VisionArena-Chat --hf-split train --num-prompts 1000
```

## Test Result

Inter-token Latency seems to improve, though I'm not sure whether this is just noise. Tested on a single L40s GPU.

### Main

```
============ Serving Benchmark Result ============
Successful requests:                     1000
Failed requests:                         0
Benchmark duration (s):                  56.35
Total input tokens:                      94327
Total generated tokens:                  120513
Request throughput (req/s):              17.74
Output token throughput (tok/s):         2138.50
Peak output token throughput (tok/s):    2713.00
Peak concurrent requests:                1000.00
Total token throughput (tok/s):          3812.33
---------------Time to First Token----------------
Mean TTFT (ms):                          28636.18
Median TTFT (ms):                        28416.69
P99 TTFT (ms):                           53127.91
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          32.46
Median TPOT (ms):                        31.65
P99 TPOT (ms):                           46.73
---------------Inter-token Latency----------------
Mean ITL (ms):                           119.67
Median ITL (ms):                         93.25
P99 ITL (ms):                            447.23
==================================================
```

### This PR

```
============ Serving Benchmark Result ============
Successful requests:                     1000
Failed requests:                         0
Benchmark duration (s):                  56.63
Total input tokens:                      94327
Total generated tokens:                  120954
Request throughput (req/s):              17.66
Output token throughput (tok/s):         2135.88
Peak output token throughput (tok/s):    4022.00
Peak concurrent requests:                1000.00
Total token throughput (tok/s):          3801.57
---------------Time to First Token----------------
Mean TTFT (ms):                          28940.85
Median TTFT (ms):                        28812.05
P99 TTFT (ms):                           53501.34
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          34.93
Median TPOT (ms):                        30.43
P99 TPOT (ms):                           61.43
---------------Inter-token Latency----------------
Mean ITL (ms):                           98.06
Median ITL (ms):                         34.30
P99 ITL (ms):                            400.73
==================================================
```

### CODE REVIEW

**Review by DarkLight1337 — APPROVED** [review]
LGTM, let's have @ywang96 confirm as well.

**Review by ywang96 — COMMENTED** [review]
I don't see too much concern of this PR but @WoosukKwon @njhill should confirm the MRv2 changes.

**Comment by DarkLight1337** [comment]
I am ok with updating the individual models accordingly, unless it occurs for most of the models.

**Review by Isotr0py — APPROVED** [review]

**Comment by DarkLight1337** [comment]
@ywang96 shall we merge this given that it solves the OOM issue?
